codersdk/deployment.go

```diff
@@ -942,6 +951,33 @@ when required by your organization's security policy.`,
 			Group:       &deploymentGroupIntrospectionPrometheus,
 			YAML:        "collect_agent_stats",
 		},
+		{
+			Name:        "Prometheus Aggregate Agent Stats By",
+			Description: fmt.Sprintf("When collecting agent stats, aggregate metrics by a given set of comma-separated labels to reduce cardinality. Accepted values are %s.", strings.Join(AcceptedMetricAggregationLabels, ", ")),
+			Flag:        "prometheus-aggregate-agent-stats-by",
+			Env:         "CODER_PROMETHEUS_AGGREGATE_AGENT_STATS_BY",
+			Value: clibase.Validate(&c.Prometheus.AggregateAgentStatsBy, func(value *clibase.StringArray) error {
+				if value == nil {
+					return nil
+				}
+
+				acceptable := make(map[string]any, len(AcceptedMetricAggregationLabels))
+				for _, label := range AcceptedMetricAggregationLabels {
+					acceptable[label] = nil
```

Comment:
should we count the occurences? in other words, is "template,username,template" a valid array?

Reply:
I wouldn't say that it's invalid enough to spit out an error tbh.

Reply:
I agree with @johnstcn, at most a notice that we're ignoring the duplicate labels but not sure if it's worth the effort.

Reply:
Agreed; I think we can leave this for now because duplicates are filtered out and it doesn't alter behaviour at all.

Reply:
Ok